.github/workflows/build_images_uv.yaml

```diff
@@ -0,0 +1,71 @@
+name: Docker Image CI
+
+on:
+  pull_request:
+    paths:
+      - "lumigator/**"
+      - ".github/**"
+    branches: ["main"]
+    # synchronized is when you push new commits
+    types: ["opened", "synchronize"]
+  push:
+    branches:
+      - main
+  # required to enable manual triggers on the GH web ui
+  workflow_dispatch:
+
+env:
+  AWS_REGION: us-east-2
+
+# https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+permissions:
+  id-token: write
+  contents: read # This is required for actions/checkout
+
+jobs:
+  build:
+    name: Build Docker image
+    runs-on: ubuntu_2204_150gb # this is a mzai-custom image on github actions runners.
+    env:
+      # head_ref: source branch of the pull request.
+      # only set on PRs; e.g., for a PR from branch `myname/amazing_model`, will be `myname/amazing_model`
+      # ref_name: short ref name of branch triggering runs. works for pushes to a development branch.
+      # assignment works starting with the head_ref
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+    # set up a local registry
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # gets full repo history
+          fetch-depth: 0
+          
+      - name: Get repo version
+        run: |
+          echo "REPOVERSION=$(git describe --tags --dirty --match \"[0-9\.]*\" --always)" >> $GITHUB_ENV
+          echo "Version of the repo for this build: $REPOVERSION"
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build the Docker image
+        run: |
+          docker build ./lumigator/python/mzai/backend/ --file ./lumigator/python/mzai/backend/Dockerfile.uv --tag mzdotai/lumigator:backend_dev_$REPOVERSION
+
+      - name: Create an image with latest tag
+        run: docker tag mzdotai/lumigator:backend_dev_$REPOVERSION mzdotai/lumigator:latest
```

Comment:
Shouldn't this be just `mzdotai/lumigator:$REPOVERSION` if we got for semver? Maybe `mzdotai/lumigator_backend:$REPOVERSION`?

Reply:
I'm good with the short tag for now

Reply:
Hey @javiermtorres, yep agree, I don't like that style of tagging. Will change it. But just to make sure that we are on the same page, $REPOVERSION is a hash. The versions tagged with semver will be those that we will release “publicly”. Are we okay with that? @chainlink @agpituk @peteski22

Reply:
Ah ok.... So this would be for dev images.... Please then say

> Create an image with latest *commit hash*

Otherwise the reader may think it's only for tags. And maybe instead of running the docker command directly, you can use the docker github actions, which make it way easier to create and tag images with git info: https://docs.docker.com/build/ci/github-actions/manage-tags-labels/

Reply:
This PR just focuses on dev versions which use the hash. We can tackle the tagged release process in another PR (when we have a release process)